hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphIndexTransaction.java

```diff
@@ -886,7 +889,22 @@ private boolean matchSearchIndexWords(String propValue, String fieldValue) {
     }
 
     private Set<String> segmentWords(String text) {
-        return this.textAnalyzer.segment(text);
+        /*
+         * Support 3 kinds of query:
+         *  - Text.contains("(word)"): query by user-specified word;
+         *  - Text.contains("word1|word2|word3"): query by user-specified words;
+         *  - Text.contains("words"): query by words splitted from analyzer;
+         * Note: all kinds support words exact match
+         */
+        if (text.startsWith(START_SYMBOL) && text.endsWith(END_SYMBOL)) {
+            return ImmutableSet.of(text.substring(1, text.length() - 1));
+        } else if (text.contains(WORD_DELIMITER)) {
+            String[] texts = StringUtils.split(text, WORD_DELIMITER);
+            return ImmutableSet.copyOf(texts);
+        }
+        Set<String> segment = this.textAnalyzer.segment(text);
+        segment.add(text);
```

Comment:
don't add text to segment

Reply:
if don't add text to segment, will can't exact match 'words'

Reply:
textAnalyzer.segment() would ensue words the same between insertion and query.